examples/ShoppingAds/GetProductBiddingCategoryConstant.php

```diff
@@ -0,0 +1,186 @@
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates the query.
+        $query = 'SELECT product_bidding_category_constant.localized_name,
+            product_bidding_category_constant.product_bidding_category_constant_parent
+            FROM product_bidding_category_constant
+            WHERE product_bidding_category_constant.country_code IN (\'US\')
+            ORDER BY product_bidding_category_constant.localized_name ASC';
+
+        // Creates a map of top level categories.
+        $rootCategories = [];
+        // Creates a map of all categories found in the results.
+        // This is a convenience lookup to enable fast retrieval of existing categories.
+        $biddingCategories = [];
+
+        // Performs the search request.
+        $response = $googleAdsServiceClient->search(
+            $customerId,
+            $query,
+            ['pageSize' => self::PAGE_SIZE]
+        );
+
+        // Iterates over all rows in all pages to extract the result.
+        foreach ($response->iterateAllElements() as $googleAdsRow) {
+            /**
+             * @var GoogleAdsRow $googleAdsRow
+             * @var ProductBiddingCategoryConstant $productBiddingCategory
+             */
+            $productBiddingCategory = $googleAdsRow->getProductBiddingCategoryConstant();
+            $resourceName = $productBiddingCategory->getResourceName();
+
+            // Adds the category in the map if new.
+            if (!array_key_exists($resourceName, $biddingCategories)) {
+                $biddingCategories[$resourceName] = [];
+            }
+            // Ensures that the localized name attribute for the categories is set.
+            // It will not be initialized for new categories or categories added
+            // to biddingCategories as a result of being a parent category below.
```

Comment:
Overall, I understand now what it means, but it was confusing at first because we don't see the below block yet.

Besides, if the $resourceName is in the $biddingCategories already, does it mean it has been added by the "parent detection" below?
if so, can't we just put the line 133 into the if clause above?
That way, we can probably do something like this:

```php
// Adds the category to the map if it's new. Some categories might be added before as a result of being the 
// parent of another categories. See the code below for detail.
if (!array_key_exists($resourceName, $biddingCategories)) {
    $biddingCategories[$resourceName] = [];
    $biddingCategories[$resourceName]['localizedName'] =
        $productBiddingCategory->getLocalizedNameUnwrapped();
}
           
```

Note that our comment is shorter and easier to grasp.

Reply:
I like your idea of set the value only when necessary. To avoid confusion, I wrapped the statement in a basic condition which makes it less confusing. LMK what you think